### DISCUSSION TITLE
how to add a default session and plugin in config.json file

### DISCUSSION THREAD

**Discussion by demis001** [GMOD/jbrowse-components#1873] [General]
How to setup a default session and plugin in config.json file? It is my first time to start using  it and quite confused with it.  Everytime I visited the link, I keep getting "start new session tab".

**Comment by cmdcolin** [comment]
You may consider using the admin-server to configure a default session

See https://jbrowse.org/jb2/docs/quickstart_gui

**Comment by demis001** [comment]
I initially  generated the reference index from a command line using this instruction from the following link ( inserted ref and a single track) https://jbrowse.org/jb2/docs/quickstart_cli.  I then attempted to setup a default session using admin-server from the link you provided. Intially, I got the admin server and pointed the url to the reference fasta index files I have generated from the command line. Got the message in green "reference generated successifully". I then exited the admin server and tested it again, I got the samething with the jbrowse logo and with three linke to reset the new session. I then restarted the the admin server again, and this time I keep getting an error.  

![error_jbrowse2](https://user-images.githubusercontent.com/8713514/113473095-71e5df80-9435-11eb-9be6-fca7e2cf2872.png)
![reset_session_bring_me_to](https://user-images.githubusercontent.com/8713514/113473100-7ad6b100-9435-11eb-9fa0-ee6a9abec406.png)

Where the session file created on the server? I want delete it from command line. 

Without the admin server, I can load the ref and single track created, but I have to manually load everytime I have visited the link, no default session. 

![without_admin_server](https://user-images.githubusercontent.com/8713514/113473333-0997fd80-9437-11eb-88ec-27e48ac9e6ad.png)

**Reply by garrettjstevens** [reply]
Sorry, I'm not sure what's going on. I've tried to reproduce this error, but have been unable to. Your configuration, including the default session, is stored in a file called "config.json" in your JBrowse directory. If you want to delete the default session, you can delete "defaultSession" from that JSON file.

If you are still having problems and your "session.json" file does not contain sensitive information, you can share that with us and we can see if we can spot any problems that way.

**Reply by rbuels** [reply]
From the screen shots above, it looks like you may be embedding jbrowse-web in another web page? Is that the case? If so, the embedded linear view might be better for that.

**Reply by demis001** [reply]
@rbuels 

No  I am not embedding it. I am using bitnami rurby stack where I can configure multiple application under the same apache2 server. Then, I configured two domain one for wordpress (https://www.seeqtl.org) and the other for jbrowse (https://jb2.seeqtl.org) through digital ocean. Then, to make a menu to go back  to home page,  I edited `index.html` under jbrowse2 to let the user back to the front page.  From wordpress, I have a menu that  point to jbrowse domain name. I have an old database built with gbrowse2 and want to replicate that with jbrowse2 or jbrowse. Till now not succeeded. 

The major thing that I want to   display  will be xyplot from gff3 with score column (-Math.log10(p-value). The plugin used above is not customizable. In gbrowse, I displayed gwas data manhattan plot with simple xyplot track that support gff3 (see the track image).  Any idea? 

![gbrowse_image](https://user-images.githubusercontent.com/8713514/114225198-096c9600-9940-11eb-9d07-5741925b3a09.png)

Any one who will guide how to generate a manhatton plot from gff3 file ? I have a lot gwas data with  gff3 format, the goal will be to display xyplot with gene label (screenshot from gbrowse track). 

![manhatan_plot](https://user-images.githubusercontent.com/8713514/114227662-471eee00-9943-11eb-9968-e139122ef95a.png)

Sample data: 

```
##gff-version 3
##Note: See http://song.sourceforge.net

# file: hg19_all_eqtl_segment_hapmap.gff
chr1	Montgomery1RNAseq_CEU_LCL	cis_eQTL_segment	40204516	40222441	-1	+	.	ID=10450_Montgomery1RNAseq_CEU_LCL;Name=PPIE
chr1	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	40219279	40219279	0.83118589907	+	.	Parent=10450_Montgomery1RNAseq_CEU_LCL;Name=PPIE;Note=rs14528;Alias=rs14528
chr1	Montgomery1RNAseq_CEU_LCL	Gene_body	40204516	40222441	-1	+	.	Parent=10450_Montgomery1RNAseq_CEU_LCL
chr10	Montgomery1RNAseq_CEU_LCL	cis_eQTL_segment	16555741	16585522	-1	-	.	ID=389941_Montgomery1RNAseq_CEU_LCL;Name=C1QL3
chr10	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	16558289	16558289	2.35209073274	-	.	Parent=389941_Montgomery1RNAseq_CEU_LCL;Name=C1QL3;Note=rs6602130;Alias=rs6602130
chr10	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	16556709	16556709	2.54012927741	-	.	Parent=389941_Montgomery1RNAseq_CEU_LCL;Name=C1QL3;Note=rs7909832;Alias=rs7909832
chr10	Montgomery1RNAseq_CEU_LCL	Gene_body	16555741	16564004	-1	-	.	Parent=389941_Montgomery1RNAseq_CEU_LCL
chr10	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	16585521	16585521	1.20861667252	-	.	Parent=389941_Montgomery1RNAseq_CEU_LCL;Name=C1QL3;Note=rs12412368;Alias=rs12412368
chr3	Montgomery1RNAseq_CEU_LCL	cis_eQTL_segment	161062579	161922679	-1	-	.	ID=165679_Montgomery1RNAseq_CEU_LCL;Name=C3orf57
chr3	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	161922678	161922678	0.792589635847	-	.	Parent=165679_Montgomery1RNAseq_CEU_LCL;Name=C3orf57;Note=rs9843284;Alias=rs9843284
chr3	Montgomery1RNAseq_CEU_LCL	Gene_body	161062579	161089871	-1	-	.	Parent=165679_Montgomery1RNAseq_CEU_LCL
chr9	Montgomery1RNAseq_CEU_LCL	cis_eQTL_segment	123544002	123652898	-1	+	.	ID=253039_Montgomery1RNAseq_CEU_LCL;Name=LOC253039
chr9	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	123610287	123610287	2.98892400381	+	.	Parent=253039_Montgomery1RNAseq_CEU_LCL;Name=LOC253039;Note=rs4837796;Alias=rs4837796
chr9	Montgomery1RNAseq_CEU_LCL	Gene_body	123605319	123616651	-1	+	.	Parent=253039_Montgomery1RNAseq_CEU_LCL
chr9	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	123640499	123640499	0.920540639754	+	.	Parent=253039_Montgomery1RNAseq_CEU_LCL;Name=LOC253039;Note=rs1953126;Alias=rs1953126
chr9	Montgomery1RNAseq_CEU_LCL	cis_eQTL_Montgomery1RNAseq_CEU_LCL	123633947	123633947	0.903233103507	+	.	Parent=253039_Montgomery1RNAseq_CEU_LCL;Name=LOC253039;Note=rs3933326;Alias=rs3933326
```


@demis001
